### ISSUE TITLE
Log transform on geom_bar() silently omits layer

### ISSUE THREAD

**Issue by krlmlr** [tidyverse/ggplot2#2147]
After some thinking, `geom_bar()` cannot work with `coord_trans(y = "log10")` (https://github.com/tidyverse/ggplot2/issues/420#issuecomment-4247826). An error message (or a warning) would be great, though.

``` r
library(tidyverse, quietly = TRUE)
diamonds %>% ggplot() + geom_bar(aes(x = cut)) + coord_trans(y = "sqrt")
```

![](http://i.imgur.com/Kp9MOSa.png)

``` r
diamonds %>% ggplot() + geom_bar(aes(x = cut)) + coord_trans(y = "log10")
```

![](http://i.imgur.com/LEMvdRn.png)

**Comment by has2k1** [comment]
I think there is a divide-by-zero in there somewhere that leads to infinite coordinate axes.

**Comment by hadley** [comment]
`log(0)` becomes -Inf, but this should generate a warning or error somewhere.

**Comment by karawoo** [comment]
Closed in #2210
